Documentation/go_client.md

```diff
@@ -0,0 +1,18 @@
+# etcd go client API
+
+This document describes tips for using the etcd go client (the package `github.com/coreos/etcd/client`).
+
+
+## Endpoint Selection Mode
+
+This section describes how to configure a policy of choosing a request destination node in etcd client programs. It will affects performance (latency and throughput) of request processing in the etcd cluster.
+
+A program that uses etcd client creates its instance of etcd client object with calling `client.New()`. The function requires an instance of `client.Config` as its parameter. The `client.Config` has a member named `SelectionMode`, typed as an enum `client.EndpointSelectionMode`. With the member, the program can specify the policy of choosing a request destination node in the etcd cluster.
+
+### EndpointSelectionRandom (default)
+
+The default value of the `SelectionMode` is `EndpointSelectionRandom`. As the name implies, the client object will pick a node from the members of the cluster in a random manner. If the cluster has three members, A, B, and C, the client picks any nodes from the three members as its request destination.
+
+### EndpointSelectionPrioritizeLeader
+
+If SelectionMode is set EndpointSelectionPrioritizeLeader, client requests directly to leader. This saves forwarding roundtrips to leader from followers. When there is a leader failure, client cannot prioritize. Therefore, EndpointSelectionPrioritizeLeader must be used with client.AutoSync() to keep the cluster configuration up-to-date.
```

Comment:
If `SelectionMode` is set to `EndpointSelectionPrioritizeLeader`, requests are sent directly to the cluster leader. This reduces forwarding roundtrips compared to making requests to etcd followers who then forward them to the cluster leader. In the event of a leader failure, however, clients configured this way cannot prioritize among the remaining etcd followers. Therefore, when a client sets `SelectionMode` to `EndpointSelectionPrioritizeLeader`, it must use `client.AutoSync()` to maintain its knowledge of current cluster state.